client/src/components/Leaflet.js

```diff
@@ -21,6 +23,25 @@ const css = {
 	zIndex: 1,
 }
 
+class CustomURLESRIProvider extends EsriProvider {
+	endpoint({ query, protocol } = {}) {
+	  const { params } = this.options;
+	  const paramString = this.getParamString({
+		...params,
+		f: 'json',
+		text: query,
+	  })
+	  return `${protocol}//${Settings.imagery.geoSearcher.url}${paramString}`;
+	}
+  }
+
+const geoSearcherProviders = {
+	ESRI : () => { return new CustomURLESRIProvider()},
+	OSM :  () => { return new OpenStreetMapProvider()}
+}
+
+const searchProvider = Settings.imagery.geoSearcher && geoSearcherProviders[Settings.imagery.geoSearcher.provider]()
```

Comment:
The lines above can be simplified to:
```
const geoSearcherProviders = {
	ESRI: CustomURLESRIProvider,
	OSM: OpenStreetMapProvider,
}

const searchProvider = Settings.imagery.geoSearcher && new geoSearcherProviders[Settings.imagery.geoSearcher.provider]()
```

Reply:
I would prefer to keep the factory as different parameters can be passed to the constructor. For instance, I should have taken the time and made the URL for ESRI to be one such parameter